packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php

```diff
@@ -48,16 +49,35 @@
             </span>
 
             @if (filled($badge))
-                <span
-                    @class([
-                        'inline-flex items-center justify-center ml-auto rtl:ml-0 rtl:mr-auto min-h-4 px-2 py-0.5 text-xs font-medium tracking-tight rounded-xl whitespace-normal',
-                        'text-primary-700 bg-primary-500/20' => ! $active,
-                        'text-white bg-white/20' => $active,
-                        'dark:text-primary-500' => (! $active) && config('filament.dark_mode'),
-                    ])
-                >
-                    {{ $badge }}
-                </span>
+                <span @class([
+                    'inline-flex items-center justify-center ml-auto rtl:ml-0 rtl:mr-auto min-h-4 px-2 py-0.5 text-xs font-medium tracking-tight rounded-xl whitespace-normal',
+                    'text-white bg-white/20' => $active,
+                    ...match ($badgeColor ?? null) {
+                        'danger' => [
+                            'text-danger-700 bg-danger-500/10' => ! $active,
+                            'dark:text-danger-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        'secondary' => [
+                            'text-gray-700 bg-gray-500/10' => ! $active,
+                            'dark:text-gray-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        'success' => [
+                            'text-success-700 bg-success-500/10' => ! $active,
+                            'dark:text-success-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        'warning' => [
+                            'text-warning-700 bg-warning-500/10' => ! $active,
+                            'dark:text-warning-500' => (! $active) && config('filament.dark_mode'),
+                        ],
+                        'primary', null => [
+                            'text-primary-700 bg-primary-500/10' => ! $active,
+                            'dark:text-primary-500' => (! $active) && config('tables.dark_mode'),
+                        ],
+                        default => [
+                            $badgeColor => ! $active,
```

Comment:
What does this do?

Reply:
Supports custom color classes

Reply:
https://github.com/filamentphp/filament/pull/3216#discussion_r928812885